docs/docsgen/source/tutorial_python/concepts.rst

```diff
@@ -0,0 +1,381 @@
+
+ONNX Concepts
+=============
+
+ONNX can be compared to a programming language specialized
+in mathematical functions. It defines all the necessary operations
+a machine learning model needs to implement its inference function
+with this langage. A linear regression could be represented
+the following way:
+
+::
+
+    def onnx_linear_regressor(X):
+        "ONNX code for a linear regression"
+        return onnx.Add(onnx.MatMul(X, coefficients), bias)
+
+.. index:: ONNX graph
+
+This example is very similar to an expression a developper could
+write in Python. It can be also represented as a graph which shows
+step by step how to transform the features to get a prediction.
+That's why a machine learning model implemented with ONNX is often
+referenced as an **ONNX graph**.
+
+.. image:: images/linreg1.png
+
+ONNX aims at providing a common language any machine learning framework
+can use to describe its models. The first scenario is to make it easier
+to deploy a machine learning model in production. An ONNX interpretor
+(or **runtime**) can be specifically implemented and optimized for this task
+in the environment where it is deployed. With ONNX, it is possible
+to build a unique process to deploy a model in production and independant
+from the learning framework used to build the model.
+
+.. contents::
+    :local:
+    :depth: 1
+
+Input, Output, Node, Initializer, Attributes
+++++++++++++++++++++++++++++++++++++++++++++
+
+Building an ONNX graph means implementing a function
+with the ONNX language or more precisely the :ref:`l-onnx-operators`.
+A linear regression would be written this way.
+The following lines do not follow python syntax.
+It is just a kind of pseudo code to illustrate the model.
+
+::
+
+    x = onnx.input(0)
```

Comment:
Why not something like
```
    Input: float[M,K] x, float[K,N] a, float[N] c
    Output: float[M, N] axc
    ax = onnx.MatMul(a, x)
    axc = onnx.Add(ax, c)
```
It's just pseudo-code. And the above is much closer to ONNX proto.